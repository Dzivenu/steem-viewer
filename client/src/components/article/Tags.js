import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { COLORS } from '../../constants/styles';

const Container = styled.div`
  display: flex;
`;

const Tag = styled.div`
  padding: 1em;
  border: 1px solid ${COLORS.WHITE.WHITE_SMOKE};
  border-radius: 1.5em;
  margin: 0 .5em;
`;

class Tags extends Component {
  static propTypes = {
    tags: PropTypes.array,
  };

  static defaultProps = {
    tags: [],
  };

  render() {
    const { tags } = this.props;
    return (
      <Container>
        {tags.map((tag, index) => <Tag key={index}>{tag}</Tag>)}
      </Container>
    );
  }
}

export default Tags;
