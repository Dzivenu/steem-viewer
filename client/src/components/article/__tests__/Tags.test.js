import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Tags from '../Tags';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('(Component) Tags', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {};
      const wrapper = shallow(<Tags {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
