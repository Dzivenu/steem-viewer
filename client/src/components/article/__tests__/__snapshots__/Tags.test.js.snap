// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Tags with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    tags={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
